PEP 690: Add flexibility to set_eager_imports() (#2571)
python/peps: pep-0690.rst

The effect of this is also shallow: all imports within ``one.mod`` will be
eager, but not imports in all modules imported by ``one.mod``.

> Or a callback object which receives a module name returns whether the import
> for such name should be eager::

Reviewer: ```suggestion
``set_eager_imports()`` can instead take a callback which receives a module name and returns
whether the import of this module should be eager::
```

> whether the import of this module should be eager::

Reviewer: We just realized in discussion that this is subtly wrong, should be fixed like so:

```suggestion
whether imports within this module should be eager::
```

> whether the import of this module should be eager::

Reviewer 2: It's unclear what "instead" is refering to in this sentence. Perhaps just set out what the function does, and then how it is different to (the other thing that 'instead' is refering to)?

A

Reviewer: I think it makes sense with the fuller context of the previous paragraphs? The immediately preceding text describes (and shows) how `set_eager_imports` works when it takes a list of module names. Then this proceeds to describe how it can instead take a callback. So I think the text already does what you recommend:

> Perhaps just set out what the function does, and then how it is different
